model/outbox.go

```diff
@@ -1,10 +1,12 @@
 package model
 
+import "github.com/google/uuid"
+
 // An OutboxEntry represents an entry in a user's AP outbox.
 type OutboxEntry struct {
 	Base
 	User       User
-	UserID     string
```

Comment:
I think this is the crux if your problem. By default, gorm will use <related model>ID as the field to store the FK value, using the value <related model>.ID as the foreign key

So the model that has a User will store the value of User.ID in UserID for the related model. This can be changed with the gorm tags foreignkey and associated_foreignkey if they are not named conventionally. Since our User has an ID of type uuid.UUID, we need to ensure UserID has type uuid.UUID.